### PR TITLE
lightwalletd: init at 0.4.7

### PR DESCRIPTION
**Pull request by centromere**
###### Motivation for this change
Add lightwalletd to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/blockchains/lightwalletd/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+buildGoModule rec {
+  pname = "lightwalletd";
+  version = "0.4.7";
+
+  src = fetchFromGitHub {
+    owner = "zcash";
+    repo  = "lightwalletd";
+    rev = "a4cc08b89eddd88befc36931422c53e4d4190185";
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/lightwalletd/default.nix:9]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/blockchains/lightwalletd/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ buildGoModule, fetchFromGitHub, lib }:
+buildGoModule rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/lightwalletd/default.nix:2]:
```suggestion
{ buildGoModule, fetchFromGitHub, lib }:

buildGoModule rec {
```

File: pkgs/applications/blockchains/lightwalletd/default.nix (hunk 1 above)
Comment (centromere) [reply pkgs/applications/blockchains/lightwalletd/default.nix]: I've made this change, however the code [explicitly](https://github.com/zcash/lightwalletd/blob/a4cc08b89eddd88befc36931422c53e4d4190185/Makefile#L22) calls for the commit hash in the `GitCommit` field.

File: pkgs/applications/blockchains/lightwalletd/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "v${version}";
+    sha256 = "0dwam3fhc4caga7kjg6cc06sz47g4ii7n3sa4j2ac4aiy21hsbjk";
+  };
+
+  vendorSha256 = null;
+
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/zcash/lightwalletd/common.Version=v${version}"
+    "-X github.com/zcash/lightwalletd/common.GitCommit=a4cc08b89eddd88befc36931422c53e4d4190185"
```
Comment (SuperSandro2000) [inline pkgs/applications/blockchains/lightwalletd/default.nix:19]: That won't get updated. Can we just set this to the version or blank instead?

Comment (centromere) [reply pkgs/applications/blockchains/lightwalletd/default.nix:19]: That's why I originally set `rev` to the commit hash, as that would enable auto-updating. I can set `GitCommit` to the version string for now and deal with it later if someone complains.

**Pull request merged**